bucket_map/src/bucket.rs

```diff
@@ -890,6 +890,152 @@ impl<'b, T: Clone + Copy + PartialEq + std::fmt::Debug + 'static> Bucket<T> {
+                    &mut entries_created,
+                    &mut duplicates,
+                    false,
+                )
+                .unwrap();
+                assert!(hashed.is_empty());
+                assert!(duplicates.is_empty());
+
+                hashed_raw.iter().for_each(|(hash, i)| {
+                    let (k, v) = raw[*i];
+                    let ix = *hash % cap;
+                    let entry = IndexEntryPlaceInBucket::new(ix);
+                    assert_eq!(entry.key(&index), &k);
+                    assert_eq!(
+                        entry.get_slot_count_enum(&index),
+                        OccupiedEnum::OneSlotInIndex(&v)
+                    );
+                });
+
+                drop(index);
+                let path = paths.first().unwrap().join(file_name.to_string());
+                let mut index = BucketStorage::<IndexBucket<u64>>::load_on_restart(
+                    path,
+                    NonZeroU64::new(
+                        std::mem::size_of::<crate::index_entry::IndexEntry<u64>>() as u64
+                    )
+                    .unwrap(),
+                    max_search,
+                    Arc::default(),
+                    Arc::default(),
+                )
+                .unwrap();
+
+                // verify index file is unoccupied, but that contents match
+                hashed_raw.iter().for_each(|(hash, i)| {
+                    let (k, _v) = raw[*i];
+                    let ix = *hash % cap;
+                    let entry = IndexEntryPlaceInBucket::new(ix);
+                    assert_eq!(entry.key(&index), &k);
+                    assert_eq!(entry.get_slot_count_enum(&index), OccupiedEnum::Free);
+                });
+
+                // this was wiped out by the last call to batch_insert..., so recreate it.
+                hashed = hashed_raw.clone();
+                let mut duplicates = Vec::default();
+                if reuse_type == 0 {
+                    Bucket::<u64>::batch_insert_non_duplicates_reusing_file(
+                        &mut index,
+                        &data_buckets,
+                        &raw,
+                        &mut hashed,
+                        &mut duplicates,
+                    );
+                } else if reuse_type == 1 {
+                    // just overwrite all data instead of trying to reuse it
+                    let mut entries_created = 0;
+                    _ = Bucket::<u64>::batch_insert_non_duplicates_internal(
+                        &mut index,
+                        &data_buckets,
+                        &raw,
+                        &mut hashed,
+                        &mut entries_created,
+                        &mut duplicates,
+                        false,
+                    );
+                    assert_eq!(entries_created, hashed_raw.len());
+                } else if reuse_type == 2 {
+                    // just overwrite all data instead of trying to reuse it
```

Comment:
nit: here we are actually reusing the disk file, right?

Reply:
yes, reusing file, but ignoring data within the file. During steady state, file can contain garbage (old data) in unoccupied entries. If the entry is unoccupied (ie. free), then the data is ignored.
This is just testing the case where we are generating the index over non-zeroed data, which could occur if the reuse insertion code fails to insert one or more entries.

Reply:
So, the comments should be updated to "trying to reuse existing data first?